nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -4461,6 +4465,62 @@ public Set<UserEntity> getUsers() {
             .collect(Collectors.toSet());
     }
 
+    /**
+     * Search for User and Group Tenants with optimized conversion from specific objects to Tenant objects
+     *
+     * @param query Search query where null or empty returns unfiltered results
+     * @return Tenants Entity containing zero or more matching Users and Groups
+     */
+    @Override
+    public TenantsEntity searchTenants(final String query) {
+        final PermissionsDTO permissions = dtoFactory.createPermissionsDto(authorizableLookup.getTenant());
+
+        final Set<TenantEntity> usersFound = userDAO.getUsers()
+                .stream()
+                .filter(user -> isMatched(user.getIdentity(), query))
+                .map(user -> createTenantEntity(user, permissions))
+                .collect(Collectors.toSet());
+
+        final Set<TenantEntity> userGroupsFound = userGroupDAO.getUserGroups()
+                .stream()
+                .filter(userGroup -> isMatched(userGroup.getName(), query))
+                .map(userGroup -> createTenantEntity(userGroup, permissions))
+                .collect(Collectors.toSet());
+
+        final TenantsEntity tenantsEntity = new TenantsEntity();
+        tenantsEntity.setUsers(usersFound);
+        tenantsEntity.setUserGroups(userGroupsFound);
+        return tenantsEntity;
+    }
+
+    private boolean isMatched(final String label, final String query) {
+        return StringUtils.isEmpty(query) || containsIgnoreCase(label, query);
+    }
+
+    private TenantEntity createTenantEntity(final User user, final PermissionsDTO permissions) {
+        final TenantDTO tenant = dtoFactory.createTenantDTO(user);
+        return createTenantEntity(tenant, permissions);
+    }
+
+    private TenantEntity createTenantEntity(final Group userGroup, final PermissionsDTO permissions) {
```

Comment:
Our `EntityFactory` already contains methods for creating a `TenantEntity` given a `tenant`, `revision`, and `permissions`. Can we use that in favor of these new methods?